src/frontend/flight/sync.rs

```diff
@@ -368,6 +381,49 @@ impl SeafowlDataSyncManager {
 
         Ok(Arc::new(UnionExec::new(vec![input_plan, proj_plan])))
     }
+
+    // Based on the provided origins and sequence numbers deduce what is the current durable
+    // sequence number for each origin.
+    fn advance_durable(
```

Comment:
Highly recommend adding some unit tests for this function, since this is where the most important logic lives and it has lots of mutability of seqs/locs/origin_durable

Reply:
I've now made a parametrizable test called `test_sync_flush` and added some cases.

Let me know what you think, and especially let me know about any cases I can add there.

Reply:
Looking good! Suggested a couple cases.